app/src/sharedTest/java/org/oppia/android/app/testing/NavigationDrawerActivityTest.kt

```diff
@@ -272,6 +273,72 @@ class NavigationDrawerActivityTest {
     }
   }
 
+  @Test
+  fun testNavigationDrawerTestActivity_openNavDrawer_clickSwitchProfile_clickCancel() {
```

Comment:
```kotlin
private fun checkNavigationViewItemStatus(item: NavigationDrawerItem) = object : TypeSafeMatcher<View>() {
    override fun describeTo(description: Description) {
      description.appendText("NavigationViewItem is checked")
    }

    override fun matchesSafely(view: View): Boolean {
      return ((view as NavigationView).menu.getItem(item.ordinal).isChecked) == true
    }
  }

  @Test
  fun testNavigationDrawerTestActivity_openNavDrawer_clickSwitchProfile_clickCancel() {
    launch<NavigationDrawerTestActivity>(
      createNavigationDrawerActivityIntent(internalProfileId)
    ).use {
      testCoroutineDispatchers.runCurrent()
      it.openNavigationDrawer()
      onView(withText(R.string.menu_switch_profile)).perform(click())
      onView(withText(R.string.home_activity_back_dialog_cancel))
        .inRoot(isDialog())
        .perform(click())
      it.openNavigationDrawer()
      testCoroutineDispatchers.runCurrent()
      onView(withId(R.id.fragment_drawer_nav_view))
        .check(matches(checkNavigationViewItemStatus(NavigationDrawerItem.HOME)))
    }
  }
```

This will work with Espresso, but might need some fix for robolectric. 
Try to fix it, and if it will not work, let me know.